media/client/main/source/MediaPipeline.cpp

```diff
@@ -566,9 +555,10 @@ void MediaPipeline::notifyNeedMediaData(int32_t sourceId, size_t frameCount, uin
 {
     RIALTO_CLIENT_LOG_DEBUG("entry:");
 
-    if (m_audioSourceSwitchOngoing && MediaSourceType::AUDIO == static_cast<MediaSourceType>(sourceId))
+    if (MediaSourceType::UNKNOWN == m_attachedSources.get(sourceId))
```

Comment:
This is kinda nice but at the same time I think there's always the edge case that the app may get a spurious needData for a removed source depending on timing & thread model so it will need some similar logic to this anyway. So there would be an argument for keeping Rialto client simple and removing this check. I don't feel strongly either way personally - what do others think?

Reply:
Rialto sink checks, if streamId is still attached before pulling sample, so I can remove this check. In such case developers of apps using rialto c++ interface will have to remember to skip attaching segments for removed source ids.

Reply:
As discussed we will keep this